### ISSUE TITLE
diffusion training loss

### ISSUE THREAD

**Issue by szlll** [jpthu17/DiffusionRet#3]
in diffusion_models/gaussian_diffusion.py 803-804
            terms["kl_loss"] = F.cross_entropy(model_output * temp,
                                               th.zeros(model_output.size(0), dtype=th.long).to(model_output.device))
I wonder why here uses the  th.zeros(model_output.size(0), dtype=th.long) instead of x_start? Is that an error?

**Comment by jpthu17** [comment]
Thank you for raising the issue. The code is correct and I will explain it below:

Specifically, we put all the positive samples in the first position of each batch.

https://github.com/jpthu17/DiffusionRet/blob/bfa5fee00b3196cca2b70733f576ea9372afea4e/DiffusionRet/models/modeling.py#L240

When calculating the loss, all samples indexed 0 are positive samples, so the label for learning is torch.zeros(model_output.size(0), dtype=th.long).

https://github.com/jpthu17/DiffusionRet/blob/bfa5fee00b3196cca2b70733f576ea9372afea4e/diffusion_models/gaussian_diffusion.py#L798-L799
